net/FlatBuffers/ByteBuffer.cs

```diff
@@ -278,13 +245,13 @@ public static int ArraySize<T>(Span<T> x)
         // the buffer position and length.
 #if ENABLE_SPAN_T
         public T[] ToArray<T>(int pos, int len)
-            where T: struct
+            where T : struct
         {
             unsafe
             {
                 AssertOffsetAndLength(pos, len);
                 T[] arr = new T[len];
-                var typed = MemoryMarshal.Cast<byte, T>(new Span<byte>(_buffer.Buffer + pos, _buffer.Length));
+                var typed = MemoryMarshal.Cast<byte, T>(_buffer.GetReadOnlySpan(pos));
                 typed.Slice(0, arr.Length).CopyTo(arr);
                 return arr;
```

Comment:
Seems like this could be simplified to just:
```C#
return MemoryMarshal.Cast<byte, T>(_buffer.GetReadOnlySpan()).Slice(pos, len).ToArray();
```
?

Reply:
It's actually a bit more complicated than that. The `pos` parameter appears to be in terms of bytes, the `len` parameter appears to be in terms of `T`.  See the "non-SPAN_T" code below, and the original code.

So the above should be:

```C#
return MemoryMarshal.Cast<byte, T>(_buffer.ReadOnlySpan.Slice(pos)).Slice(0, len).ToArray();
```

Reply:
Yup, you're right.